web_src/js/features/imagediff.js

```diff
@@ -98,174 +101,184 @@ export function initImageDiff() {
+      imgParent.style.margin = `0px ${sizes.ratio[0] * factor}px`;
+      imgParent.style.width = `${sizes.sizeAfter.width * factor + 2}px`;
+      imgParent.style.height = `${sizes.sizeAfter.height * factor + 2}px`;
+      swipeFrame.style.padding = `${sizes.ratio[1] * factor}px 0 0 0`;
+      swipeFrame.style.width = `${sizes.maxSize.width * factor + 2}px`;
+    }
 
-      $container.find('.swipe-bar').on('mousedown', function(e) {
-        e.preventDefault();
+    if (sizes.imageBefore) {
+      const imgParent = sizes.imageBefore.parentNode;
+      const swipeFrame = imgParent.parentNode;
+      sizes.imageBefore.style.width = `${sizes.sizeBefore.width * factor}px`;
+      sizes.imageBefore.style.height = `${sizes.sizeBefore.height * factor}px`;
+      imgParent.style.margin = `${sizes.ratio[3] * factor}px ${sizes.ratio[2] * factor}px`;
+      imgParent.style.width = `${sizes.sizeBefore.width * factor + 2}px`;
+      imgParent.style.height = `${sizes.sizeBefore.height * factor + 2}px`;
+      swipeFrame.style.width = `${sizes.maxSize.width * factor + 2}px`;
+      swipeFrame.style.height = `${sizes.maxSize.height * factor + 2}px`;
+    }
 
-        const $swipeBar = $(this);
-        const $swipeFrame = $swipeBar.parent();
-        const width = $swipeFrame.width() - $swipeBar.width() - 2;
+    // extra height for inner "position: absolute" elements
+    const swipe = this.containerEl.querySelector('.diff-swipe');
+    if (swipe) {
+      swipe.style.width = `${sizes.maxSize.width * factor + 2}px`;
+      swipe.style.height = `${sizes.maxSize.height * factor + 30}px`;
+    }
 
-        $(document).on('mousemove.diff-swipe', (e2) => {
-          e2.preventDefault();
+    this.containerEl.querySelector('.swipe-bar').addEventListener('mousedown', (e) => {
+      e.preventDefault();
+      this.initSwipeEventListeners(e.currentTarget);
+    });
+  }
 
-          const value = Math.max(0, Math.min(e2.clientX - $swipeFrame.offset().left, width));
-          $swipeBar[0].style.left = `${value}px`;
-          $container.find('.swipe-container')[0].style.width = `${$swipeFrame.width() - value}px`;
+  initSwipeEventListeners(swipeBar) {
+    const swipeFrame = swipeBar.parentNode;
+    const width = swipeFrame.clientWidth;
+    const onSwipeMouseMove = (e) => {
+      e.preventDefault();
+      const rect = swipeFrame.getBoundingClientRect();
+      const value = Math.max(0, Math.min(e.clientX - rect.left, width));
+      swipeBar.style.left = `${value}px`;
+      this.containerEl.querySelector('.swipe-container').style.width = `${swipeFrame.clientWidth - value}px`;
+    };
+    const removeEventListeners = () => {
+      document.removeEventListener('mousemove', onSwipeMouseMove);
+      document.removeEventListener('mouseup', removeEventListeners);
+    };
+    document.addEventListener('mousemove', onSwipeMouseMove);
+    document.addEventListener('mouseup', removeEventListeners);
```

Comment:
1. What happened to the previous `.diff-swipe` part of the event listener selector? I don't see a test for this class anymore in the new event listener being tested anymore. Does that have any implications, or is that condition now no longer necessary?
2. Do I see it correctly that this listener is only called once? Is that intended?

Reply:
1. It is a jQuery feature "event namespace" https://api.jquery.com/event.namespace/
2. Why once? They will be added again when "mousedown"

Reply:
This is the old logic. I didn't rewrite it to avoid arguments like "why the event listeners are left here while they were removed in old code by jQuery off"

Reply:
Ah, there it is.
I was already wondering as the code seemed strange with an "only once" listener.